### PR TITLE
jetbrains.plugins: update

### PR DESCRIPTION
**Pull request by dritter**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I don't know how to test this, hence this is just a draft PR. Maybe that fixes https://github.com/NixOS/nixpkgs/pull/240521#issuecomment-1627648790

But as @pbek in https://github.com/NixOS/nixpkgs/pull/240521#issuecomment-1627703897 I don't know how to build/test this locally.

// cc @Janik-Haag

### CODE REVIEW

**Comment by dritter** [comment]
Result of `nixpkgs-review pr 242493` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Review by pbek — APPROVED** [review]
🥳 I'm using this commit in my config now. I can confirm GitHub copilot is working in phpstorm and clion! Thank you very much! 🥳 

PhpStorm 2023.1.3
Build #PS-231.9161.47, built on June 22, 2023

**Comment by Janik-Haag** [comment]
Then we should make the jetbrains update script call the plugin update script. @GenericNerdyUsername what do you think?

**Comment by GenericNerdyUsername** [comment]
> Then we should make the jetbrains update script call the plugin update script. @GenericNerdyUsername what do you think?

Might have to be careful about the plugin updater script using a module that the ide updater doesnt (so you got a `ModuleNotFoundError` bc it isnt in the nix-shell shebang), but other than that, seems like a perfectly good solution

**Comment by Janik-Haag** [comment]
Okay I think we merge this so upstream isn't broken anymore and someone opens a draft pr that makes the jetbrains update script also update the plugins so it doesn't break again in the future?

**Comment by pbek** [comment]
If anyone else needs to test, this is what I did: https://github.com/pbek/nixcfg/blob/68fe04422c5b8cc8df22ad0481776eceb12421a6/modules/mixins/jetbrains.nix#L3-L21

**Comment by Janik-Haag** [comment]
Who where want's to / has the time to open the pr for updating the update script?

**Comment by Janik-Haag** [comment]
Also I'm not quite sure about the commit name, since usually it's like $versionA -> $versionB but that's not really possible in this case.

**Comment by dritter** [comment]
> Who where want's to / has the time to open the pr for updating the update script?

I can prepare a PR in the next days, if noone else wants.

> Also I'm not quite sure about the commit name, since usually it's like $versionA -> $versionB but that's not really possible in this case.

I agree. This makes finding a good commit message way harder. The current one is the best I could think of. If you have a different suggestion, I can rephrase it.

**Comment by Janik-Haag** [comment]
> I can prepare a PR in the next days, if noone else wants.

Sounds good to me

> I agree. This makes finding a good commit message way harder. The current one is the best I could think of. If you have a different suggestion, I can rephrase it.

I would at least at like a time-stamp when the update script ran (can probably be patched into the update script) to make it easier finding commits.

**Comment by dritter** [comment]
> I would at least at like a time-stamp when the update script ran (can probably be patched into the update script) to make it easier finding commits.

Alright. I included a timestamp manually for now. Maybe we should include the timestamp somewhere in the `plugins.json`.

**Comment by GenericNerdyUsername** [comment]
What if we did something like `nix flake update --commit-lock-file`?

File: pkgs/applications/editors/jetbrains/plugins/plugins.json [hunk 1]
```diff
@@ -153,11 +153,10 @@
...
       "builds": {
-        "231.9011.34": "https://plugins.jetbrains.com/files/8554/326468/featuresTrainer-231.8770.66.zip",
-        "231.9011.35": "https://plugins.jetbrains.com/files/8554/326468/featuresTrainer-231.8770.66.zip",
-        "231.9011.38": "https://plugins.jetbrains.com/files/8554/326468/featuresTrainer-231.8770.66.zip",
-        "231.9011.41": "https://plugins.jetbrains.com/files/8554/326468/featuresTrainer-231.8770.66.zip",
-        "231.9161.38": "https://plugins.jetbrains.com/files/8554/326468/featuresTrainer-231.8770.66.zip"
+        "231.9161.29": "https://plugins.jetbrains.com/files/8554/326468/featuresTrainer-231.8770.66.zip",
+        "231.9161.38": "https://plugins.jetbrains.com/files/8554/326468/featuresTrainer-231.8770.66.zip",
+        "231.9161.40": "https://plugins.jetbrains.com/files/8554/326468/featuresTrainer-231.8770.66.zip",
+        "231.9161.41": "https://plugins.jetbrains.com/files/8554/326468/featuresTrainer-231.8770.66.zip"
```
Comment (Janik-Haag) [inline pkgs/applications/editors/jetbrains/plugins/plugins.json:159]: Here is one more deletion then additon is that because a plugin was removed?

**Comment by Janik-Haag** [comment]
Okay one last thing some one suggested we could do `jetbrains.plugins: update` which I think sounds good and other updates like vimPlugins or gnomeExtensions also don't give any further information so I would change it to that. Sorry for all the trouble. 

I'll merge this If you answer the question above and change the commit message a last time. Sorry :see_no_evil:

File: pkgs/applications/editors/jetbrains/plugins/plugins.json (hunk 1 above)
Comment (GenericNerdyUsername) [reply pkgs/applications/editors/jetbrains/plugins/plugins.json:159]: different ide builds support different plugin builds, which sometimes means you go from each ide having it's own version of a plugin to two ides supporting the same plugin, so the total amount of plugins supported drops by one

**Comment by Janik-Haag** [comment]
I took the liberty of renaming and will merge once ci was successful,

**Comment by dritter** [comment]
Thanks. Sorry for being unresponsive, it has been a busy day..

**Comment by Janik-Haag** [comment]
No worries I just didn't want to cause any further delay since upstream is currently broken.

**Comment by pbek** [comment]
@dritter, @Janik-Haag, @GenericNerdyUsername
After updating my flake to the current NixOS unstable, I got the error:

```
clion-with-plugins> sed: can't read /nix/store/nzi9z8l4f8viqsm1npq9q68q13ks5x9l-clion-with-plugins-2023.1.4/bin/.clion-wrapped: No such file or directory
error: builder for '/nix/store/ahdk3nnwiif1pm7gj3lz97pg67nzz5ln-clion-with-plugins-2023.1.4.drv' failed with exit code 2
error: 1 dependencies of derivation '/nix/store/3kp0vv2dj9plrhgr2fmi9y2m13dg2rig-clion-with-plugins-2023.1.4_fish-completions.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ydd790yw28vdx8sdcp9xpq64kpk77xl6-man-paths.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fajl1wk8cxvkcghi7xbdx0fdhjs20ggh-system-path.drv' failed to build
```

(config: https://github.com/pbek/nixcfg/blob/72f0331775f85aac4f1b4e30162a720c43b15ae5/modules/mixins/jetbrains.nix#L20>

Did anyone else have this?

**Comment by dritter** [comment]
@pbek haven't looked deeper Info this, but I guess it has something to do with https://github.com/NixOS/nixpkgs/pull/243533

**Comment by pbek** [comment]
Hm, https://github.com/NixOS/nixpkgs/pull/243533 doesn't seem to be in unstable yet: https://nixpk.gs/pr-tracker.html?pr=243533

**Comment by GenericNerdyUsername** [comment]
sorry, that was probable me [here](https://github.com/NixOS/nixpkgs/pull/242724/commits/dd4054b93dd7653483e3f446ef1269bfe96b643a), ill get started on a fix and some tests straight away

**Pull request merged**